### PR TITLE
[UFC-1187] Support authentication token in sign-in JS action

### PR DESCRIPTION
**Pull request by EnasAbdelrazek**
## Checklist

-   Contains unit tests: ❌
-   Contains breaking changes ❌
-   Compatible with: MX 10.12
-   Did you update version and changelog? ✅
-   PR title properly formatted (`[XX-000]: description`)? ✅ 
-   Works in Android ✅ 
-   Works in iOS ✅ 
-   Works in Tablet ✅ 

## This PR contains

-   [ ] Bug fix
-   [x] Feature
-   [ ] Refactor
-   [ ] Documentation
-   [ ] Other (describe)

## What is the purpose of this PR?
Update the sign-in JS action to support the authentication token feature that allows users to remain signed in until the expiration time of the token.

## Extra comments (optional)
To release this module: we should update the Sign-in JS action in the nanoflow-commons module to include the new parameter useAuthToken

### CODE REVIEW

File: packages/jsActions/nanoflow-actions-native/src/client/SignIn.ts [hunk 1]
```diff
@@ -32,7 +33,11 @@ export async function SignIn(username?: string, password?: string): Promise<Big>
         const onSuccess = (): void => resolve(new Big(200));
         const onError = (error: { status: number }): void => resolve(new Big(error.status));
 
-        mx.login(username, password, onSuccess, onError as any);
+        if (typeof useAuthToken === "undefined") {
+            mx.login(username, password, onSuccess, onError as any);
+        } else {
+            mx.login2(username, password, Boolean(useAuthToken), onSuccess, onError as any);
```
Comment (Robinfr) [inline packages/jsActions/nanoflow-actions-native/src/client/SignIn.ts:39]: Why do you cast it to `Boolean`? It should already be a boolean right?

File: packages/jsActions/nanoflow-actions-native/typings/mx.d.ts [hunk 2]
```diff
@@ -11,5 +11,12 @@ declare namespace mx {
     }
     interface MxInterface {
         reload: () => void;
+        login2(
```
Comment (Robinfr) [inline packages/jsActions/nanoflow-actions-native/typings/mx.d.ts:14]: Why is this needed? I don't see the original `login` here either 😕

File: packages/jsActions/nanoflow-actions-native/src/client/SignIn.ts (hunk 1 above)
Comment (EnasAbdelrazek) [reply packages/jsActions/nanoflow-actions-native/src/client/SignIn.ts:39]: Yes, that is right.

File: packages/jsActions/nanoflow-actions-native/typings/mx.d.ts (hunk 2 above)
Comment (EnasAbdelrazek) [reply packages/jsActions/nanoflow-actions-native/typings/mx.d.ts:14]: login exists in mendix-client typings.

Comment (Robinfr) [reply packages/jsActions/nanoflow-actions-native/typings/mx.d.ts:14]: Yeah I don't know where that package comes from to be honest, but got it.

**Comment by Robinfr** [comment]
Good to go from our side 👍

File: packages/jsActions/nanoflow-actions-native/typings/mx.d.ts (hunk 2 above)
Comment (EnasAbdelrazek) [reply packages/jsActions/nanoflow-actions-native/typings/mx.d.ts:14]: It was published 5 years ago :D https://www.npmjs.com/package/mendix-client

**Review by Ahmad-Elsayed — APPROVED** [review]

**Review by Robinfr — APPROVED** [review]
